iio/iio_ad7124/iio_ad7124.c

```diff
@@ -0,0 +1,849 @@
+		size_t len, const struct iio_ch_info *channel, intptr_t priv);
+static ssize_t ad1724_iio_read_raw_chan(void *device, char *buf, size_t len,
+					const struct iio_ch_info *channel, intptr_t priv);
+static ssize_t ad1724_iio_read_odr_chan(void *device, char *buf, size_t len,
+					const struct iio_ch_info *channel, intptr_t priv);
+static ssize_t ad1724_iio_change_odr_chan(void *device, char *buf, size_t len,
+		const struct iio_ch_info *channel, intptr_t priv);
+static ssize_t ad1724_iio_read_scale_chan(void *device, char *buf, size_t len,
+		const struct iio_ch_info *channel, intptr_t priv);
+static ssize_t ad1724_iio_change_scale_chan(void *device, char *buf, size_t len,
+		const struct iio_ch_info *channel, intptr_t priv);
+
+static struct iio_attribute channel_attributes[] = {
+	{
+		.name = "filter_low_pass_3db_frequency",
+		.priv = 0,
+		.show = ad1724_iio_read_filter_3db,
+		.store = ad1724_iio_write_filter_3db
+	},
+	{
+		.name = "offset",
+		.priv = 0,
+		.show = ad1724_iio_read_offset_chan,
+		.store = ad1724_iio_change_offset_chan
+	},
+	{
+		.name = "raw",
+		.priv = 0,
+		.show = ad1724_iio_read_raw_chan,
+		.store = NULL
+	},
+	{
+		.name = "sampling_frequency",
+		.priv = 0,
+		.show = ad1724_iio_read_odr_chan,
+		.store = ad1724_iio_change_odr_chan
+	},
+	{
+		.name = "scale",
+		.priv = 0,
+		.show = ad1724_iio_read_scale_chan,
+		.store = ad1724_iio_change_scale_chan
+	},
+	END_ATTRIBUTES_ARRAY
+};
+
+struct scan_type ad7124_iio_scan_type = {
+	.sign = 'i',
+	.realbits = 24,
+	.storagebits = 32,
+	.shift = 0,
+	.is_big_endian = true
+};
+
+#define AD7124_IIO_CHANN_DEF(nm, ch1, ch2) \
+	{ \
+		.name = nm, \
+		.ch_type = IIO_VOLTAGE, \
+		.channel = ch1, \
+		.channel2 = ch2, \
+		.scan_type = &ad7124_iio_scan_type, \
+		.attributes = channel_attributes, \
+		.ch_out = 0, \
+		.indexed = 1, \
+		.diferential = true, \
+	}
+
+static struct iio_channel ad7124_channels[] = {
+	AD7124_IIO_CHANN_DEF("ch0", 0, 1),
+	AD7124_IIO_CHANN_DEF("ch1", 2, 3),
+	AD7124_IIO_CHANN_DEF("ch2", 4, 5),
+	AD7124_IIO_CHANN_DEF("ch3", 6, 7),
+	AD7124_IIO_CHANN_DEF("ch4", 8, 9),
+	AD7124_IIO_CHANN_DEF("ch5", 10, 11),
+	AD7124_IIO_CHANN_DEF("ch6", 12, 13),
+	AD7124_IIO_CHANN_DEF("ch7", 14, 15)
+};
+
+/**
+ * @brief Read and display channel offset.
+ * @param device - Device driver descriptor.
+ * @param buf - Output buffer.
+ * @param len - Length of the input buffer (not used in this case).
+ * @param channel - IIO channel information.
+ * @return Number of bytes printed in the output buffer, or negative error code.
+ */
+static ssize_t ad1724_iio_read_offset_chan(void *device, char *buf, size_t len,
+		const struct iio_ch_info *channel, intptr_t priv)
+{
+	struct ad7124_dev	*desc = (struct ad7124_dev *)device;
+	uint32_t		value;
+	int32_t ret;
+
+	ret = ad7124_read_register(desc,
+				   &desc->regs[(AD7124_OFFS0_REG + channel->ch_num)]);
+	if (ret != SUCCESS)
+		return ret;
+	value = desc->regs[(AD7124_OFFS0_REG + channel->ch_num)].value;
```

Comment:
I think fields from inside desc should pe only accesed inside the drivere that created that structure. So my approach will be to create in the driver a function ad7124_read_offset_chan . 
Same for other the remaining functions

Reply:
And then you can do a switch calling that function like this: https://github.com/analogdevicesinc/no-OS/blob/05cd411971cb9368179cdadbf695d6a95f404ba3/drivers/platform/aducm3029/iio_aducm3029.c#L222